wasm2js0.test_autodebug_wasm regression

```

C:\emsdk\emscripten\main>test\runner wasm2js0.test_autodebug_wasm
Running test_core: (1 tests)
test_autodebug_wasm (test_core.wasm2js0.test_autodebug_wasm) ... env_modify: {'EMCC_AUTODEBUG': '1'}
-- begin program output --
Aborted(Assertion failed: missing Wasm export: __get_temp_ret)
C:\emsdk\emscripten\main\out\test\test_autodebug.js:5893
  var e = new WebAssembly.RuntimeError(what);
          ^

Error: Aborted(Assertion failed: missing Wasm export: __get_temp_ret)
    at abort (C:\emsdk\emscripten\main\out\test\test_autodebug.js:5893:11)
    at assert (C:\emsdk\emscripten\main\out\test\test_autodebug.js:5593:5)
    at assignWasmExports (C:\emsdk\emscripten\main\out\test\test_autodebug.js:7042:3)
    at receiveInstance (C:\emsdk\emscripten\main\out\test\test_autodebug.js:5998:5)
    at receiveInstantiationResult (C:\emsdk\emscripten\main\out\test\test_autodebug.js:6019:12)
    at createWasm (C:\emsdk\emscripten\main\out\test\test_autodebug.js:6045:17)
Thrown at:
    at abort (C:\emsdk\emscripten\main\out\test\test_autodebug.js:5893:11)
    at assert (C:\emsdk\emscripten\main\out\test\test_autodebug.js:5593:5)
    at assignWasmExports (C:\emsdk\emscripten\main\out\test\test_autodebug.js:7042:3)
    at receiveInstance (C:\emsdk\emscripten\main\out\test\test_autodebug.js:5998:5)
    at receiveInstantiationResult (C:\emsdk\emscripten\main\out\test\test_autodebug.js:6019:12)
    at createWasm (C:\emsdk\emscripten\main\out\test\test_autodebug.js:6045:17)


Node.js v22.16.0
-- end program output --
FAIL

======================================================================
FAIL: test_autodebug_wasm (test_core.wasm2js0.test_autodebug_wasm)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\emsdk\emscripten\main\test\common.py", line 1000, in resulting_test
    return func(self, *args)
  File "C:\emsdk\emscripten\main\test\common.py", line 717, in metafunc
    return func(self, *args, **kwargs)
  File "C:\emsdk\emscripten\main\test\test_core.py", line 267, in decorated
    return func(self, *args, **kwargs)
  File "C:\emsdk\emscripten\main\test\common.py", line 502, in modified
    return func(self, *args, **kwargs)
  File "C:\emsdk\emscripten\main\test\test_core.py", line 7050, in test_autodebug_wasm
    output = self.do_runf('core/test_autodebug.c', 'success')
  File "C:\emsdk\emscripten\main\test\common.py", line 2057, in do_runf
    return self._build_and_run(filename, expected_output, **kwargs)
           ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\emsdk\emscripten\main\test\common.py", line 2105, in _build_and_run
    js_output = self.run_js(js_file, engine, args,
                            input=input,
                            assert_returncode=assert_returncode,
                            interleaved_output=interleaved_output)
  File "C:\emsdk\emscripten\main\test\common.py", line 1684, in run_js
    self.fail('JS subprocess failed (%s): %s (expected=%s).  Output:\n%s' % (error.cmd, error.returncode, assert_returncode, ret))
    ~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: JS subprocess failed (C:/emsdk/node/22.16.0_64bit/bin/node.exe --stack-trace-limit=50 --trace-uncaught C:\emsdk\emscripten\main\out\test\test_autodebug.js): 1 (expected=0).  Output:
Aborted(Assertion failed: missing Wasm export: __get_temp_ret)
C:\emsdk\emscripten\main\out\test\test_autodebug.js:5893
  var e = new WebAssembly.RuntimeError(what);
          ^

Error: Aborted(Assertion failed: missing Wasm export: __get_temp_ret)
    at abort (C:\emsdk\emscripten\main\out\test\test_autodebug.js:5893:11)
    at assert (C:\emsdk\emscripten\main\out\test\test_autodebug.js:5593:5)
    at assignWasmExports (C:\emsdk\emscripten\main\out\test\test_autodebug.js:7042:3)
    at receiveInstance (C:\emsdk\emscripten\main\out\test\test_autodebug.js:5998:5)
    at receiveInstantiationResult (C:\emsdk\emscripten\main\out\test\test_autodebug.js:6019:12)
    at createWasm (C:\emsdk\emscripten\main\out\test\test_autodebug.js:6045:17)
Thrown at:
    at abort (C:\emsdk\emscripten\main\out\test\test_autodebug.js:5893:11)
    at assert (C:\emsdk\emscripten\main\out\test\test_autodebug.js:5593:5)
    at assignWasmExports (C:\emsdk\emscripten\main\out\test\test_autodebug.js:7042:3)
    at receiveInstance (C:\emsdk\emscripten\main\out\test\test_autodebug.js:5998:5)
    at receiveInstantiationResult (C:\emsdk\emscripten\main\out\test\test_autodebug.js:6019:12)
    at createWasm (C:\emsdk\emscripten\main\out\test\test_autodebug.js:6045:17)


Node.js v22.16.0


----------------------------------------------------------------------
Ran 1 test in 1.206s

FAILED (failures=1)

C:\emsdk\emscripten\main>
```

bisecting Emscripten repository points to #25541 as the regressing commit.

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -717,6 +717,7 @@ jobs:
             core2s.test_dylink_syslibs_missing_assertions
             core2s.test_module_wasm_memory
             cores.test_minimal_runtime_safe_heap
+            wasm2js0.test_autodebug_wasm
             wasm2js3.test_memorygrowth_2
             wasm2js2.test_pthread_proxying
             wasmfs.test_hello_world

--- a/test/test_core.py
+++ b/test/test_core.py
@@ -7017,9 +7017,6 @@ void* operator new(size_t size) {
     if '-msimd128' in self.cflags:
       self.skipTest('https://github.com/emscripten-core/emscripten/issues/25001')
 
-    if self.is_wasm2js() and '-O0' in self.cflags:
-      self.skipTest('https://github.com/emscripten-core/emscripten/issues/25549')
-
     # Even though the test itself doesn't directly use reference types,
     # Binaryen's '--instrument-locals' will add their logging functions if
     # reference-types is enabled. So make sure this test passes when

